src/app/user.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from './shared/services/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
```

Comment:
@olivierobert Actually it comes with the code when I generated the guard with the command `ng g guard ...` but regard [this document about DI](https://angular.io/guide/dependency-injection) it makes the class visible to the whole application so it can be created and uses everywhere 🤔